src/der.rs

```diff
@@ -107,6 +107,50 @@ pub(crate) fn read_tag_and_get_value<'a>(
     ring::io::der::read_tag_and_get_value(input).map_err(|_| Error::BadDer)
 }
 
+#[inline(always)]
+#[allow(dead_code)]
+pub(crate) fn read_tag_and_get_value_limited<'a>(
+    input: &mut untrusted::Reader<'a>,
+) -> Result<(u8, untrusted::Input<'a>), Error> {
+    fn read_byte(input: &mut untrusted::Reader) -> Result<u8, Error> {
```

Comment:
After this first commit, should we do some refactoring to make this code nicer? I would suggest:

* Implement `impl From<untrusted::EndOfInput> for Error`, after which we can inline this wrapper
* Maybe name the binary consts here?
* Perhaps rewrite this so that it can scale to longer values without deduplicating the match arms

Reply:
The first two sound reasonable to me. I implemented both in my recent push. Some of the constants are a little bit wordy but it's a tricky domain to capture succinctly. 

> Perhaps rewrite this so that it can scale to longer values without deduplicating the match arms? (I guess that might be bad for performance, actually?)

I went back/forth on the duplication of the arms when I first wrote this code. I also felt like it may be bad for performance and I don't think we'll be adding any more match arms in the future. The duplication is a little bit gross but it leaves the code easy to read (especially with the new constants). I also peeked at the [rust-asn1 crate](https://github.com/alex/rust-asn1) to see what it did and it looks like there might be some shared heredity (or there's just not many ways to write this kind of code!). It also has [duplicative `match` arms](https://github.com/alex/rust-asn1/blob/1d10d020f9b5cc7a21295791beb596b942d77449/src/parser.rs#L212-L257) very closely resembling our own.

My inclination is to leave it as-is for now, but I can revisit if you have stronger feelings.